messages2/src/decorators/thread.rs

```diff
@@ -0,0 +1,86 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize, Serializer};
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
```

Comment:
I think this can be just `Eq` (i know it was `PartialEq` in original messages crate, but see no reason no to make it `Eq`, or am I missing something?)

Reply:
Why would we implement `Eq`? The `PartialEq` is mostly for tests really, so we can easily do assertions.

`Eq` is a marker trait that essentially says "this type has some additional properties" ([docs](https://doc.rust-lang.org/beta/std/cmp/trait.Eq.html)).

Most of the time `Eq` is not needed, unless you're implementing some sort of primitive and need to consider the properties outlined by the trait.

Reply:
It's not needed, but it just seem reasonable that message equality is reflexive, symmetrical and transitive. I can't imagine why that would not hold. 
`Eq` implies `PartialEq`, but not the other way around, so the fact that only `PartialEq` raises in me a question: "eh does equivalency not hold here? Why would that be so?"

Not a big deal though, consider this and feel free to mark this conversation as Resolved.

Reply:
I feel it's more of a YAGNI thing.